### ISSUE TITLE
bug(web): setRect nosize and nomove do not work according to docs

### ISSUE THREAD

**Issue by mcdurdin** [keymanapp/keyman#4947]
**Describe the bug**

`keyman.osk.setRect()` does not appear to work as the [documentation](https://help.keyman.com/developer/engine/web/14.0/reference/osk/setRect) describes. According to the documentation, the following should stop the user moving and sizing the OSK:
```js
keyman.osk.setRect({nosize:'none',nomove:'none'});
```

And the following should make it possible for the user to move and size the OSK again:
```js
keyman.osk.setRect({nosize:'block',nomove:'block'});
```

However, from the code, `nosize` and `nomove` are booleans. Testing shows that passing `true`/`false` for these parameters produces the expected behaviour. (Side note: the parameter names are inverted. It is better to use positive parameter names. I would love to see more meaningful parameter names -- `nosize` should really be `allowUserSize` or something like that, and obviously the value would be inverse).

A secondary issue is that once `setRect({left:x, top:y})` has been called, `restorePosition()` no longer resets the position to pin to the active text field. Internally, this is because once `keyman.osk.dfltX` and `keyman.osk.dfltY` are set, there is no API to reset them. `restorePosition()` should probably be clearing those values (or there should be another API to do that).

I would also like to deprecate the `setPos()` function in favour of the `setRect()` function... It's confusing having both. Perhaps even extending `setRect()` to support an additional parameter which is `followActiveTextField: boolean`?

---

**KeymanWeb:**
 - OS: any 
 - Browser: any 
 - KeymanWeb Version: 14.0.271

**Comment by jahorton** [comment]
> **Describe the bug**
> 
> `keyman.osk.setRect()` does not appear to work as the [documentation](https://help.keyman.com/developer/engine/web/14.0/reference/osk/setRect) describes. According to the documentation, the following should stop the user moving and sizing the OSK:
> 
> ```js
> keyman.osk.setRect({nosize:'none',nomove:'none'});
> ```
> 
> And the following should make it possible for the user to move and size the OSK again:
> 
> ```js
> keyman.osk.setRect({nosize:'block',nomove:'block'});
> ```
> 
> However, from the code, `nosize` and `nomove` are booleans. Testing shows that passing `true`/`false` for these parameters produces the expected behaviour. (Side note: the parameter names are inverted. It is better to use positive parameter names. I would love to see more meaningful parameter names -- `nosize` should really be `allowUserSize` or something like that, and obviously the value would be inverse).

I'd like to point out that this has been the case since KMW 10.0:

https://github.com/keymanapp/keyman/blob/f16872a99896a2fc208c68ac92a827820ab07e3f/web/source/kmwosk.ts#L454-L464

The mild restructuring of that code was done as part of KMW 12.0:

https://github.com/keymanapp/keyman/blob/3148aa7b8c8a4b5de574d49bdf6edc80dcb6761a/web/source/osk/oskManager.ts#L1182-L1196

From stable-14.0:

https://github.com/keymanapp/keyman/blob/ec9a68cd2422bdfc0f0d7513c6eeb6a9b1d85b5a/web/source/osk/oskManager.ts#L1147-L1161

So... the documentation's been off for half-of-forever.

**Comment by mcdurdin** [comment]
> So... the documentation's been off for half-of-forever.

So? We still need to fix it.

**Comment by jahorton** [comment]
> I would also like to deprecate the `setPos()` function in favour of the `setRect()` function... It's confusing having both. Perhaps even extending `setRect()` to support an additional parameter which is `followActiveTextField: boolean`?

Inspecting the code... there is a distinction worth noting here:

- `setPos`:
    - does _not_ set `dfltX` or `dfltY`
    - interpretation:  intended to temporarily relocate the position of the OSK
- `setRect`:
    - does set `dfltX` and `dfltY`
    - other parameters seem designed for page-specific OSK display customization
        - `nosize`, `nomove`:
            - disables certain forms of user interaction that can alter OSK display properties
    - interpretation:  intended to customize a page's preferences for a user's interaction with the OSK as part of page init

Furthermore, `dfltX` and `dfltY` are not persisted to KMW's cookies.  This is likely due to the assumption that they'd be automatically re-established during a page's initiation on future visits / refreshes.

> So?  We still need to fix it.

Right.  The documentation.

**Comment by jahorton** [comment]
The one possible conflict with that interpretation:  `setPos` doesn't provide the ability to alter the OSK's width and height.

**Comment by mcdurdin** [comment]
> This is likely due to the assumption that they'd be automatically re-established during a page's initiation on future visits / refreshes.

Right. We just need a way to do that reset during a page's lifetime so it's not a dead end call 😁

> The one possible conflict with that interpretation: `setPos` doesn't provide the ability to alter the OSK's width and height.

It still feels messy -- surely we can find a way to integrate the calls so we no longer need `setPos()`.



> Right. The documentation. If it's _just_ that, this issue is tagged on the wrong repo - documentation is part of help.keyman.com's domain.

This should probably be split into two separate issues. When I started investigating, it wasn't clear to me where the problem lay. So I will move this issue to help and create a new one about `setRect` not providing a path to clear `dfltX` and `dfltY`.
